importccl: update parquet vendor

msbutler opened the pull request:
importccl: update parquet vendor to v0.10.0
    
Release note: None

Release justification: bug fixes and low-risk updates to new functionality

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76689)
<!-- Reviewable:end -->

msbutler committed:
importccl: update parquet vendor to v0.10.0

Release note: None

Release justification: bug fixes and low-risk updates to new functionality

stevendanna commented on DEPS.bzl:
```diff
@@ -2448,10 +2458,10 @@ def go_deps():
-        sha256 = "66beb8f2218c31ca4ca3d3dbcc91c256f3379750ade924016f4179982446edd7",
-        strip_prefix = "github.com/fraugster/parquet-go@v0.6.1",
+        sha256 = "68aa69311763a3c816552da6b2571272f996bc111237cecbb171dd7d41bb9346",
+        strip_prefix = "github.com/fraugster/parquet-go@v0.8.0",
```
Looks like the latest is 0.9.0, want to bump so close to their most recent release, but figured I'd point it out.

msbutler replied:
yeah, there's a known regression in 0.9.0 they're working on :/

Hope to update to latest release during stability.

stevendanna reviewed (approved):
LGTM.

Sometimes we include the highlights of or links to the changelog for the changes in the commit message.

msbutler commented:
TFTR!

bors r=stevendanna

msbutler commented:
bors r=stevendanna
